src/main/kotlin/org/vaccineimpact/api/models/Expectations.kt

```diff
@@ -14,31 +17,70 @@ data class Expectations(
                 && (cohorts.maximumBirthYear == null || this <= cohorts.maximumBirthYear)
     }
 
-    fun expectedRows(): List<ExpectedRow>
-    {
-        val listRows = mutableListOf<ExpectedRow>()
+    private fun outcomesMap() = outcomes.associateBy({ it }, { null })
+
+    fun expectedCentralRows(disease: String): Sequence<ExpectedCentralRow> = buildSequence{
         for (age in ages)
         {
             for (country in countries)
             {
-                listRows.addAll(years
-                        .map { it - age }
-                        .filter { it.withinCohortRange() }
-                        .map { ExpectedRow(country.id, age, it) })
+                yieldAll(years
+                        .filter { (it - age).withinCohortRange() }
+                        .map { mapCentralRow(disease, it, age, country) })
             }
         }
+    }
 
-        return listRows
+    fun expectedStochasticRows(disease: String): Sequence<ExpectedStochasticRow> = buildSequence {
+        for (age in ages)
+        {
+            for (country in countries)
+            {
+                yieldAll(years
+                        .filter { (it - age).withinCohortRange() }
+                        .map { mapStochasticRow(disease, it, age, country) })
+            }
+        }
```

Comment:
Feels like more of the code could be shared between `expectedStochasticRows` and `expectedCentralRows`?

Reply:
I played around with that, but couldn't find a nice way to both share the code, and keep the return types distinguished. In the end I thought a bit of repetitive logic was worth it for clearly distinguished methods with their own type signatures.

Reply:
How about:

```
data class RowDeterminer(year: Int, age: Int, country: Country)

fun expectedCentralRows(diease: String) = expectedRows()
	.map { mapCentralRow(disease, it.year, it.age, it.country) }

fun expectedStochasticRows(diease: String) = expectedRows()
	.map { mapStochasticRow(disease, it.year, it.age, it.country) }

private fun expectedRows(): Sequence<RowDeterminer> = buildSequence {
    for (age in ages)
    {
        for (country in countries)
        {
        	yieldAll(years
        		     .filter { (it - age).withinCohortRange() }
                     .map { RowDeterminer(it, age, country) })

        }
    }
}
```
fun expectedCentralRows(diease: String) = expectedRows()
	.map { mapCentralRow(disease, it.year, it.age, it.country) }

fun expectedStochasticRows(diease: String) = expectedRows()
	.map { mapStochasticRow(disease, it.year, it.age, it.country) }
```

Reply:
That works, thanks :)